### PR TITLE
feat: allow relative file paths

### PR DESCRIPTION
**Pull request by FUSAKLA**
Signed-off-by: Martin Chodur <m.chodur@seznam.cz>

@seznam/sklik-devops-sre 

Allow file paths to be relative.

@dohnto @rudo-thomas FYI

### CODE REVIEW

File: template.go [hunk 1]
```diff
@@ -68,14 +68,6 @@ func generateTemplate(source, name string, delimLeft string, delimRight string)
 }
 
 func generateFile(templatePath, destinationPath string, debugTemplates bool, delimLeft string, delimRight string) error {
-	if !filepath.IsAbs(templatePath) {
```
Comment (dohnto) [inline template.go:71]: Can we make some tests covering this functionality?

Comment (FUSAKLA) [reply template.go:71]: Sure thing, PTAL https://github.com/seznam/goenvtemplator/pull/25/commits/06358b5d42275b9cad65d393ebb36eea1e899b88

**Comment by FUSAKLA** [comment]
@dohnto PTAL I refactored the test file handling

File: template_test.go [hunk 2]
```diff
@@ -2,13 +2,77 @@ package main
 
 import (
 	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
```
Comment (rudo-thomas) [inline template_test.go:5]: FWIW, assert's "MIT license" *is* compatible with goenvtemplator's LGPL3.

File: template_test.go [hunk 3]
```diff
@@ -2,13 +2,77 @@ package main
...
+	_, err = tmpFile.WriteString(content)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return rel, abs, func() {
+		path := abs
+		err := os.Remove(path)
+		if err != nil {
+			return
+		}
```
Comment (rudo-thomas) [inline template_test.go:39]: I don't see why we need a separate variable or the no-op `if`.

```suggestion
		_ := os.Remove(abs)
```

**Review by rudo-thomas — CHANGES_REQUESTED** [review]
LGTM, but there are a few things...

File: template_test.go (hunk 2 above)
Comment (FUSAKLA) [reply template_test.go:5]: THX for checking!
cr-secutiry

File: template_test.go (hunk 3 above)
Comment (FUSAKLA) [reply template_test.go:39]: Yep, I was previously handling that error, so this is just a leftover from some refactoring.
Accepting the suggestion :+1:

**Review by rudo-thomas — APPROVED** [review]

**Pull request merged**